aliyun/src/main/java/org/apache/iceberg/aliyun/AliyunClientFactory.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aliyun;
+
+import com.aliyun.oss.OSS;
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.iceberg.common.DynConstructors;
+
+public interface AliyunClientFactory extends Serializable {
+  /**
+   * Create an aliyun OSS client.
+   *
+   * @return oss client.
+   */
+  OSS oss();
+
+  /**
+   * Initialize Aliyun client factory from catalog properties.
+   *
+   * @param properties catalog properties
+   */
+  void initialize(Map<String, String> properties);
+
+  /**
+   * Returns an initialized {@link AliyunProperties}
+   */
+  AliyunProperties aliyunProperties();
+
+  static AliyunClientFactory load(Map<String, String> properties) {
+    String impl = properties.getOrDefault("client.factory", DefaultAliyunClientFactory.class.getName());
```

Comment:
I think we will need to provide the KMS approach to manage the access key and access secret in future.
https://github.com/aliyun/aliyun-sdk-managed-credentials-providers-java

Reply:
ok. I would like to implement this functionality in a separated PR later.